currency_monthly_rate/tests/test_currency_monthly_rate.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import TransactionCase
+from odoo import fields
+
+
+class TestCurrencyMonthlyRate(TransactionCase):
+
+    def setUp(self):
```

Comment:
@grindtildeath seems you are creating several records here: can you move it to SavepointCase + setUpClass? Is going to be much more efficient.